packages/ripple-nuxt-tide/modules/preview/templates/plugin.js

```diff
@@ -1,9 +1,18 @@
-export default function ({ $axios, app, redirect }) {
+import { logger } from '@dpc-sdp/ripple-nuxt-tide/lib/core'
+
+export default function ({ $auth, $axios }) {
+
+  $auth.onError((error, name, endpoint) => {
+    if (process.server) {
+      logger.debug('Auth error: "%O"', name, { error, label: 'Tide preview' })
+    }
+  })
 
   $axios.onError(error => {
-    // Redirect to login page again if both token and refresh token have expired.
-    if (error.name?.includes('ExpiredAuthSessionError')) {
-      redirect('/oauth/login')
+    // Sometime BE will return login page instead of json error response if token is invalid
+    if (error.response?.status === 401 && error.response?.data?.error === undefined) {
+      logger.debug('Aixos error with no proper error, e.g HTML login page', { error, label: 'Tide preview' })
```

Comment:
Does it have to be always a 401 and undefined error?
Also *Axios (:  and should this be Axios returned unauthorised error or something since it is 401?

Reply:
Tide should give error response but in some API request, Tide gives a redirect to login page with 401 error, instead of oauth error. I saw some discuss for that before, I think that is a thing hard to be changed in Tide for now. So this is a patch for that issue.